UefiCpuPkg/Library/MpInitLib/MpLib.c

```diff
@@ -1621,14 +1621,30 @@ TimedWaitForApFinish (
                               TimeLimit,
                               &CpuMpData->CurrentTime
                               );
-  while (CpuMpData->FinishedCount < FinishedApLimit &&
-         !CheckTimeout (
-            &CpuMpData->CurrentTime,
-            &CpuMpData->TotalTime,
-            CpuMpData->ExpectedTime
-            ))
   {
-    CpuPause ();
+    UINTN  LoopCount;
+    UINTN  PauseCount;
+    UINTN  Index;
+
+    LoopCount  = 0;
+    PauseCount = 1;
+
+    while (CpuMpData->FinishedCount < FinishedApLimit &&
+           !CheckTimeout (
+              &CpuMpData->CurrentTime,
+              &CpuMpData->TotalTime,
+              CpuMpData->ExpectedTime
+              ))
+    {
+      for (Index = 0; Index < PauseCount; Index++) {
+        CpuPause ();
+      }
+
+      LoopCount++;
+      if ((LoopCount % 8) == 0) {
+        PauseCount *= 2;
+      }
+    }

```

Comment:
PauseCount, in an extreme case, will be a very large value.
How about limit PauseCount to a max value such as 1024?

Reply:
Ok, done @niruiyu